### ISSUE TITLE
[BUG] "Your session crashed after using all available RAM"

### ISSUE THREAD

**Issue by Expertium** [open-spaced-repetition/fsrs4anki#338]
**Describe the bug**
![image](https://github.com/open-spaced-repetition/fsrs4anki/assets/83031600/0d0b2349-b757-40cf-b2ce-21ad3d8c6bb7)
I get this error when using mini-batch.ipynb.

**To Reproduce**
1. Go to mini-batch.ipynb
2. Run the optimizer
3. Wait until it gets to section 2.5 "Predict memory states and distribution of difficulty"

**Environment**
 - Optimizer version: [mini-batch](https://github.com/open-spaced-repetition/fsrs4anki/blob/350b52319f4dce74c67ceeb53fae13e2395d640e/experiment/mini-batch.ipynb)

**Additional context**
My collection: https://drive.google.com/file/d/1YDiSjlr_FvnUG-vl5e9OTmeAfyuLaDFN/view?usp=sharing

**Comment by L-M-Sherlock** [comment]
It seems like the optimizer costs too much RAM. I will try to reduce the RAM usage.

**Comment by L-M-Sherlock** [comment]
Does this error also occur in the stable version?

**Comment by Expertium** [comment]
I'm not at home right now, so I can't test it. You can test it yourself since I provided a link to my collection.

**Comment by L-M-Sherlock** [comment]
I have located the error:

```python
stabilities, difficulties = my_collection.batch_predict(dataset)
```

This line of code will predict all data in your collection in parallel, which cost 10+GB RAM. I will optimize it.

**Comment by L-M-Sherlock** [comment]
This issue has been solved in #340 and https://github.com/open-spaced-repetition/fsrs4anki/commit/9cff568b949166f77eace655208d7e1cc48985c8
